transport-native-epoll/src/test/java/io/netty/channel/epoll/ManualEventLoopTest.java

```diff
@@ -0,0 +1,252 @@
+                    }
+                    try {
+                        taskSubmitted.await();
+                    } catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+        };
+        waitBeforeSubmittingTask.await();
+        CountDownLatch completed = new CountDownLatch(1);
+        group.ioEventLoopRunner.execute(completed::countDown);
+        taskSubmitted.countDown();
+        completed.await();
+        group.shutdownGracefully(0, 0, TimeUnit.SECONDS).get();
+    }
+
+    @RepeatedTest(value = 100, failureThreshold = 1)
+    void testTightShutodownEpoll() throws InterruptedException, ExecutionException {
+        ManualMultithreadedIoEventLoopGroup group = new ManualMultithreadedIoEventLoopGroup(
+                EpollIoHandler.newFactory());
+        group.shutdownGracefully(0, 0, TimeUnit.SECONDS).get();
+    }
+
+    @RepeatedTest(value = 100, failureThreshold = 1)
+    void testTightShutodownNIO() throws InterruptedException, ExecutionException {
+        ManualMultithreadedIoEventLoopGroup group = new ManualMultithreadedIoEventLoopGroup(
+                NioIoHandler.newFactory());
+        group.shutdownGracefully(0, 0, TimeUnit.SECONDS).get();
+    }
+
+    @Test
+    void testTightSubmissionsEpoll() throws ExecutionException, InterruptedException {
+        ManualMultithreadedIoEventLoopGroup group = new ManualMultithreadedIoEventLoopGroup(
+                EpollIoHandler.newFactory());
+        for (int i = 0; i < 1000; i++) {
+            group.submit(() -> {
+            }).get();
+        }
+        group.shutdownGracefully(0, 0, TimeUnit.SECONDS).get();
+    }
+
+    @Test
+    void testTightSubmissionsNIO() throws ExecutionException, InterruptedException {
+        ManualMultithreadedIoEventLoopGroup group = new ManualMultithreadedIoEventLoopGroup(
+                NioIoHandler.newFactory());
+        for (int i = 0; i < 1000; i++) {
+            group.submit(() -> {
+            }).get();
+        }
+        group.shutdownGracefully(0, 0, TimeUnit.SECONDS).get();
+    }
+
+    public static class ManualMultithreadedIoEventLoopGroup extends MultiThreadIoEventLoopGroup {
+
+        private ManualIoEventLoopRunner ioEventLoopRunner;
+        private Consumer<ManualIoEventLoopRunner> beforeCanBlock;
+
+        public ManualMultithreadedIoEventLoopGroup(IoHandlerFactory ioHandlerFactory) {
+            super(1, ioHandlerFactory);
+        }
+
+        @Override
+        protected IoEventLoop newChild(Executor executor, IoHandlerFactory ioHandlerFactory, Object... args) {
+            this.ioEventLoopRunner = new ManualIoEventLoopRunner(this, ioHandlerFactory,
+                    executor, this::beforeCanBlock);
+            return ioEventLoopRunner.ioEventLoop;
+        }
+
+        protected void beforeCanBlock(Executor executor) {
+        }
+
+        private static class ManualIoEventLoopRunner implements Executor {
```

Comment:
@normanmaurer this way to use the manual event loop looks correct/idiomatic in your opinion?

It's basically a simplified version of https://github.com/franz1981/Netty-VirtualThread-Scheduler enabling the manual event loop to be used to consume/handle other type of tasks than the task ones.